test/unit/trades-utils.spec.ts

```diff
@@ -0,0 +1,219 @@
+import { IPgComponent } from '@well-known-components/pg-component'
+import { formatEther } from 'ethers'
+import {
+  Network,
+  ChainId,
+  Trade,
+  TradeAssetType,
+  TradeType,
+  ERC721TradeAsset,
+  Events,
+  Rarity,
+  NFTCategory,
+  ERC20TradeAsset,
+  WearableCategory,
+  CollectionItemTradeAsset
+} from '@dcl/schemas'
+import { IEventPublisherComponent } from '../../src/ports/events'
+import { getItemByItemIdQuery } from '../../src/ports/items/queries'
+import { DBItem } from '../../src/ports/items/types'
+import { getNftByTokenIdQuery } from '../../src/ports/nfts/queries'
+import { DBNFT } from '../../src/ports/nfts/types'
+import { triggerEvent } from '../../src/ports/trades/utils'
+
+describe('when calling triggerEvent function', () => {
+  let mockPgComponent: IPgComponent
+  let mockEventPublisherComponent: IEventPublisherComponent
+  let mockPgQuery: jest.Mock
+  let mockPublishMessage: jest.Mock
+  let trade: Trade
+
+  beforeEach(() => {
+    mockPgQuery = jest.fn()
+    mockPublishMessage = jest.fn()
+
+    mockPgComponent = {
+      query: mockPgQuery
+    } as unknown as IPgComponent
+
+    mockEventPublisherComponent = {
+      publishMessage: mockPublishMessage
+    }
+
+    trade = {
+      id: '1',
+      createdAt: Date.now(),
+      signature: '0xsignature',
+      signer: '0xsigner',
+      chainId: ChainId.ETHEREUM_SEPOLIA,
+      type: TradeType.BID,
+      checks: {
+        effective: Date.now(),
+        expiration: Date.now() + 1000000,
+        allowedRoot: '0x',
+        contractSignatureIndex: 0,
+        signerSignatureIndex: 0,
+        externalChecks: [],
+        salt: '0x',
+        uses: 1
+      },
+      network: Network.ETHEREUM,
+      sent: [
+        {
+          assetType: TradeAssetType.ERC20,
+          contractAddress: '0x9d32aac179153a991e832550d9f96441ea27763a',
+          extra: '0x',
+          amount: '100'
+        }
+      ],
+      received: [
+        {
+          assetType: TradeAssetType.ERC721,
+          contractAddress: '0x9d32aac179153a991e832550d9f96441ea27763b',
+          tokenId: 'atokenid',
+          extra: '0x',
+          beneficiary: '0x9d32aac179153a991e832550d9f96441ea27763b'
+        }
+      ]
+    }
+  })
+
+  describe('when inserted trade is a bid', () => {
+    describe('when bidded asset is an nft', () => {
+      let nftBid: Trade
+      let dbNFT: DBNFT
+
+      beforeEach(async () => {
```

Comment:
The `async` directive can be removed here and in the `beforeEach` down below as well:
```suggestion
      beforeEach(() => {
```